### PR TITLE
Resize options

### PR DESCRIPTION
**Pull request by celiasmith**
Added an option in the configure modal to choose between having fixed aspect ratio objects on resizing of the canvas or not (e.g. when you open/close the editor).

### CODE REVIEW

**Comment by tcstewar** [comment]
With @jgosmann 's comments addressed, this looks fairly good to me.  I do think the default should be to have this be off, though, as I find it is much slower to update than the old method, leading to jerky resizing.

**Comment by celiasmith** [comment]
@tcstewar actually, I intended to leave it off by default (thanks for the catch).  However, I don't think I get a difference in jerkiness. (There is little difference in the number of calculations; basically, three multiplies per item, which is tiny compared to what's already going on).  That said, the graphs in particular do move differently in a way that looks less continuous on smooth resizes, I'd agree.  Switching between edit and non edit seems equally smooth.  Anyway, we can see what folks think!

**Comment by jgosmann** [comment]
I don't get a difference in jerkiness either.

**Comment by tcstewar** [comment]
> There is little difference in the number of calculations; basically, three multiplies per item, which is tiny compared to what's already going on)

You're calling ```NetGraphItem.set_size()``` inside a loop in ```NetGraph.on_resize()```.  That's extremely expensive, since ```set_size``` has to reposition both the item and all of the connections into and out of the item, and do the same thing to any children inside the item.  And since it's inside a loop, this means that in any large network a whole bunch of things are going to end up getting repositioned multiple times.

**Comment by celiasmith** [comment]
Sorry, just pushed, thought i did that yesterday...

File: nengo_gui/static/netgraph.js [hunk 1]
```diff
@@ -625,3 +640,11 @@ Nengo.NetGraph.prototype.scaleMiniMapViewBox = function () {
     this.view.setAttribute('width', w / this.scale);
     this.view.setAttribute('height', h / this.scale);
 }
+
+Nengo.NetGraph.prototype.get_aspect_resize = function() {
```
Comment (tbekolay) [inline nengo_gui/static/netgraph.js]: Is the getter / setter necessary here? Not using them might be more clear.

Comment (tcstewar) [inline nengo_gui/static/netgraph.js]: Hmm..  we should make a general style decision here.  I could definitely see generally making setters/getters even when we're not doing anything, for the standard reasons of making it easier to make them complex in the future.

Comment (tbekolay) [inline nengo_gui/static/netgraph.js]: These are apparently a builtin feature of Javascript. See [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#Defining_getters_and_setters). So, similar to Python, we should use raw attributes until we need something complex, at which point we can write this style of getter / setter without changing the code that's using those attributes.

Comment (tcstewar) [inline nengo_gui/static/netgraph.js]: Sweet!  I had no idea Janascript supports that... I totally agree.

**Comment by tbekolay** [comment]
I rebased this and added a commit to fix up the `calc_axes_geometry` stuff (hopefully?) This can use some testing to make sure it's working post-rebase.

**Comment by tbekolay** [comment]
One issue I noticed with this: resizing no longer moves graphs, only netgraph items. Seems to be due to removing the getters and setters; fixing now.

**Comment by tbekolay** [comment]
Kay, fixed (we missed a call to `get_aspect_resize()`). If someone could test this branch and give a +1, I'll merge!

**Comment by jgosmann** [comment]
Only gave it a short test, but seems to work.

**Comment by tcstewar** [comment]
That seems to work for me, too.  It's still noticeably slower for complex displays for me, but that can be resolved in the future.

**Comment by tcstewar** [comment]
That fixes the speed for me.

**Comment by celiasmith** [comment]
cool, could someone merge this?

**Comment by tbekolay** [comment]
Right, doing so now!

**Pull request merged**